src/main/java/com/amazon/sns/javamessaging/AmazonSNSExtendedClientBase.java

```diff
@@ -0,0 +1,420 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.sns.javamessaging;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.AmazonWebServiceRequest;
+import com.amazonaws.ResponseMetadata;
+import com.amazonaws.regions.Region;
+import com.amazonaws.services.sns.AmazonSNS;
+import com.amazonaws.services.sns.model.*;
+
+import java.util.List;
+
+abstract class AmazonSNSExtendedClientBase implements AmazonSNS {
+    private final AmazonSNS amazonSnsToBeExtended;
+
+    AmazonSNSExtendedClientBase(AmazonSNS snsClient) {
+        amazonSnsToBeExtended = snsClient;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public AddPermissionResult addPermission(String queueUrl, String label, List<String> aWSAccountIds, List<String> actions)
+            throws AmazonClientException {
+        return amazonSnsToBeExtended.addPermission(queueUrl, label, aWSAccountIds, actions);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public ResponseMetadata getCachedResponseMetadata(AmazonWebServiceRequest request) {
+        return amazonSnsToBeExtended.getCachedResponseMetadata(request);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void setEndpoint(String endpoint) throws IllegalArgumentException {
+        amazonSnsToBeExtended.setEndpoint(endpoint);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void setRegion(Region region) throws IllegalArgumentException {
+        amazonSnsToBeExtended.setRegion(region);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void shutdown() {
+        amazonSnsToBeExtended.shutdown();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public AddPermissionResult addPermission(AddPermissionRequest addPermissionRequest) {
+        return amazonSnsToBeExtended.addPermission(addPermissionRequest);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public CheckIfPhoneNumberIsOptedOutResult checkIfPhoneNumberIsOptedOut(CheckIfPhoneNumberIsOptedOutRequest checkIfPhoneNumberIsOptedOutRequest) {
+        return amazonSnsToBeExtended.checkIfPhoneNumberIsOptedOut(checkIfPhoneNumberIsOptedOutRequest);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public ConfirmSubscriptionResult confirmSubscription(ConfirmSubscriptionRequest confirmSubscriptionRequest) {
+        return amazonSnsToBeExtended.confirmSubscription(confirmSubscriptionRequest);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public ConfirmSubscriptionResult confirmSubscription(String s, String s1, String s2) {
```

Comment:
Can we use the parameter names from the http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/sns/AmazonSNS.html#publish-com.amazonaws.services.sns.model.PublishRequest- 

Those are a lot more descriptive:

confirmSubscription(String topicArn, String token, String authenticateOnUnsubscribe)